### PR TITLE
Popover and Tooltip cleanup

### PR DESCRIPTION
**Pull request by masvelio**
## Link to Issue
Closes: #5504 

## Description of Changes
- We had mess in the code related to the Tooltip and Popover components. We used Tooltip and Popover nomenclature interchangeably. Also, along the way, we introduced new and old Tooltip (former black, latter white) on top of Popover. This PR tidies things up thanks to multiple steps taken:  
- `CWTooltip` - component that displays one (max two) lines of text content - black component 
image
- `CWPopover` - component that displays bigger content, not only text based. Might be also used as a foundation for popover menu. 
image
image
- adjusted `CWPopover` to new design system specification (added title)
- removed storybook code related to tooltips and popovers
- removed old implementation of tooltip - `cw_tooltip`
- removed some of the dead code and dead components

## Test Plan
- Click around the app, and make sure that tooltips and popovers look and work good 
- For example
  - reactions
  - share popover
  - notifications popover
  - three dots menus
  - tooltip for thread/comment action buttons (eg when you are not part of the community)
  - top navbar tooltips
  - etc

## Deployment Plan

1. no requirements

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/LockWithTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { CWIcon } from './component_kit/cw_icons/cw_icon';
+
+type LockWithTooltip = {
+  lockedAt: moment.Moment;
+  updatedAt?: moment.Moment;
+};
+
+export const LockWithTooltip = ({ lockedAt, updatedAt }: LockWithTooltip) => {
+  return (
+    <>
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/LockWithTooltip.tsx:14]: the `<>` seems redundant

**Review by mzparacha — COMMENTED** [review]
Code review done, LGTM. Will do the local testing next...

**Review by lzach83 — COMMENTED** [review]
Code looks good, pulling it down now.

File: packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/AuthorAndPublishInfo/AuthorAndPublishInfo.tsx [hunk 2]
```diff
@@ -202,6 +201,8 @@ export const AuthorAndPublishInfo = ({
 
       {isHot && <CWTag iconName="trendUp" label="Trending" type="trending" />}
 
+      {isSpamThread && <CWTag label="SPAM" type="disabled" />}
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/discussions/ThreadCard/AuthorAndPublishInfo/AuthorAndPublishInfo.tsx:204]: 🙏

**Review by mzparacha — APPROVED** [review]
Tested locally, LGTM

**Review by lzach83 — APPROVED** [review]
Tested, works as intended

File: packages/commonwealth/client/scripts/views/components/LockWithTooltip.tsx (hunk 1 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/components/LockWithTooltip.tsx]: removed

**Pull request merged**